src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1119,17 +1120,21 @@ public Solution WithProjectAnalyzerReferences(ProjectId projectId, IEnumerable<A
 
         private Solution AddDocument(DocumentState state)
         {
-            if (state == null)
+            var notificationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
+            using (notificationService?.Start("Remove Document"))
```

Comment:
How does this actually work? Does it tell the solution crawler to permanently ignore any workspace change events during this time? If so, then at the end of this process how does solution crawler know to go re-analyze everything now that we're done (there will be no workspace change event to trigger it). Or is it just a batcher that gathers up the events during global operations and then handles them all at once in the end? If so, then what's to stop it from triggering between each of these mini-batches (unless we move it up to the project-level load/unload somehow).